src/main/java/uk/ac/ebi/intact/psi/mi/xmlmaker/utils/VersionUtils.java

```diff
@@ -0,0 +1,225 @@
+package uk.ac.ebi.intact.psi.mi.xmlmaker.utils;
+
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+import static uk.ac.ebi.intact.psi.mi.xmlmaker.utils.GuiUtils.*;
+
+
+/**
+ * Utility class for managing the versioning and update functionality of the PSI-MI XML Maker application.
+ * <p>
+ * This class checks for newer versions of the application on GitHub, downloads the appropriate artifact based on
+ * the operating system, and optionally schedules a restart to complete the update process.
+ * </p>
+ */
+public class VersionUtils {
+
+    private static final Logger LOGGER = Logger.getLogger(VersionUtils.class.getName());
+//    private static final String VERSION_URL = "https://raw.githubusercontent.com/MICommunity/psi-mi-xml-maker/refs/heads/main/src/main/resources/xmlMaker.properties"; //todo: change for release
+    private static final String VERSION_URL = "https://raw.githubusercontent.com/MICommunity/psi-mi-xml-maker/refs/heads/1.1/src/main/resources/xmlMaker.properties";
```

Comment:
Remember to update the version URL to the main branch when we're ready to merge.